### PR TITLE
PAYARA-3214 Fixing sonar blocker bugs in module common utilities

### PR DESCRIPTION
**Pull request by svendiedrichsen**
https://sonarcloud.io/project/issues?id=org.glassfish.main%3Aglassfish-main-aggregator&moduleUuids=AWJyK9if1B6WA4TgV4EE&resolved=false&severities=BLOCKER&types=BUG

### CODE REVIEW

File: appserver/pom.xml [hunk 1]
```diff
@@ -142,7 +142,7 @@
         <jms-api.version>2.0.1</jms-api.version>
 
         <!-- 5.1.1-b02.payara-p1 -->
-        <mq.version>5.1.1.final.payara-p4</mq.version>
+        <mq.version>5.1.2-SNAPSHOT</mq.version>
```
Comment (Cousjava) [inline appserver/pom.xml]: This should be left at 5.1.1.final.payara-p4

Comment (svendiedrichsen) [reply appserver/pom.xml]: Sure. That seems to be branch bleeding. Sorry for this and thanks for the catch.

**Comment by Cousjava** [comment]
Jenkins test please

**Comment by arjantijms** [comment]
Jenkins test please

**Comment by MarkWareham** [comment]
Jenkins test please
I want to review failure, but job not available

**Comment by arjantijms** [comment]
Jenkins test please

**Comment by svendiedrichsen** [comment]
I'd love to help with this. There seems to be something triggered by my PR but I can't access the server logs. Wouldn't it be nice to automatically attach server logs after build has finished?

**Comment by svendiedrichsen** [comment]
I'd love to help with this. There seems to be something triggered by my PR but I can't access the server logs. Wouldn't it be nice to automatically attach server logs after build has finished?

**Comment by svendiedrichsen** [comment]
I'd love to help with this. There seems to be something triggered by my PR but I can't access the server logs. Wouldn't it be nice to automatically attach server logs after build has finished?

**Comment by svendiedrichsen** [comment]
I'd love to help with this. There seems to be something triggered by my PR but I can't access the server logs. Wouldn't it be nice to automatically attach server logs after build has finished?

**Comment by svendiedrichsen** [comment]
I'd love to help with this. There seems to be something triggered by my PR but I can't access the server logs. Wouldn't it be nice to automatically attach server logs after build has finished?

**Comment by svendiedrichsen** [comment]
I'd love to help with this. There seems to be something triggered by my PR but I can't access the server logs. Wouldn't it be nice to automatically attach server logs after build has finished?

**Comment by MarkWareham** [comment]
Jenkins is missing the test job. Parent job has marked build job as success, test as failure.

**Comment by svendiedrichsen** [comment]
@arjantijms I'd advise against forcing the merge. I'd rather analyse the problem first.

**Comment by Pandrex247** [comment]
I shall have a look locally now and see if I can reproduce what's failing.

**Comment by Pandrex247** [comment]
Worked for me locally, I'll run it through Jenkins again.

**Comment by Pandrex247** [comment]
Jenkins test please

**Comment by svendiedrichsen** [comment]
@MarkWareham The test states that there was a test failure during deployment:
```
remote failure: Error occurred during deployment: Stream closed. Please see server.log for more details. [testng] [testng] FAILED: simpleOSGIDeployTest [testng] java.lang.AssertionError: App deployment failed ... expected:<true> but was:<false>
```
Can we take a look at the server.log file of the corresponding deployment?

**Comment by Pandrex247** [comment]
Whoops yep forgot to run quicklook locally, I can recreate it locally.
Blergh, OSGi errors.

**Comment by MarkWareham** [comment]
Looking for the server.log now

**Comment by Pandrex247** [comment]
Going to ninja as I have it locally.
I believe this is the log extract relevant to this failing test:
```
[2018-10-29T16:55:41.245+0000] [Payara 5.184-SNAPSHOT] [INFO] [] [org.glassfish.osgiweb] [tid: _ThreadID=46 _ThreadName=admin-thread-pool::admin-listener(1)] [timeMillis: 1540832141245] [levelValue: 800] [[
  u = [webbundle:], proto = [webbundle], host = [], port = [-1], auth = [null], user = [null], path = [file:/home/andrew/Git/Payara/appserver/tests/quicklook/dist/basicosgi/osgitest.war], query = [Bundle-SymbolicName=bar&Import-Package=javax.servlet;javax.servlet.http&Web-ContextPath=/osgitest], ref = [Bundle-SymbolicName=bar&Import-Package=javax.servlet;javax.servlet.http&Web-ContextPath=/osgitest]]]
[2018-10-29T16:55:41.247+0000] [Payara 5.184-SNAPSHOT] [INFO] [] [org.glassfish.osgiweb] [tid: _ThreadID=46 _ThreadName=admin-thread-pool::admin-listener(1)] [timeMillis: 1540832141247] [levelValue: 800] [[
  new path = [file:/home/andrew/Git/Payara/appserver/tests/quicklook/dist/basicosgi/osgitest.war], new query = [Bundle-SymbolicName=bar&Import-Package=javax.servlet;javax.servlet.http&Web-ContextPath=/osgitest]]]
[2018-10-29T16:55:41.254+0000] [Payara 5.184-SNAPSHOT] [INFO] [NCLS-OSGI-00001] [javax.enterprise.osgi.container] [tid: _ThreadID=46 _ThreadName=admin-thread-pool::admin-listener(1)] [timeMillis: 1540832141254] [levelValue: 800] [[
  Decorated url = webbundle:file:/home/andrew/Git/Payara/appserver/tests/quicklook/dist/basicosgi/osgitest.war?Bundle-SymbolicName=bar&Import-Package=javax.servlet;javax.servlet.http&Web-ContextPath=/osgitest]]
[2018-10-29T16:55:41.268+0000] [Payara 5.184-SNAPSHOT] [SEVERE] [] [] [tid: _ThreadID=46 _ThreadName=admin-thread-pool::admin-listener(1)] [timeMillis: 1540832141268] [levelValue: 1000] [[
  Manifest-Version: 1.0
]]
[2018-10-29T16:55:41.268+0000] [Payara 5.184-SNAPSHOT] [SEVERE] [] [] [tid: _ThreadID=46 _ThreadName=admin-thread-pool::admin-listener(1)] [timeMillis: 1540832141268] [levelValue: 1000] [[
  Ant-Version: Apache Ant 1.9.4
]]
[2018-10-29T16:55:41.269+0000] [Payara 5.184-SNAPSHOT] [SEVERE] [] [] [tid: _ThreadID=46 _ThreadName=admin-thread-pool::admin-listener(1)] [timeMillis: 1540832141269] [levelValue: 1000] [[
  Bundle-SymbolicName: bar
]]
[2018-10-29T16:55:41.269+0000] [Payara 5.184-SNAPSHOT] [SEVERE] [] [] [tid: _ThreadID=46 _ThreadName=admin-thread-pool::admin-listener(1)] [timeMillis: 1540832141269] [levelValue: 1000] [[
  Bundle-ClassPath: WEB-INF/classes
]]
[2018-10-29T16:55:41.269+0000] [Payara 5.184-SNAPSHOT] [SEVERE] [] [] [tid: _ThreadID=46 _ThreadName=admin-thread-pool::admin-listener(1)] [timeMillis: 1540832141269] [levelValue: 1000] [[
  Web-ContextPath: /osgitest
]]
[2018-10-29T16:55:41.269+0000] [Payara 5.184-SNAPSHOT] [SEVERE] [] [] [tid: _ThreadID=46 _ThreadName=admin-thread-pool::admin-listener(1)] [timeMillis: 1540832141269] [levelValue: 1000] [[
  Bundle-ManifestVersion: 2
]]
[2018-10-29T16:55:41.269+0000] [Payara 5.184-SNAPSHOT] [SEVERE] [] [] [tid: _ThreadID=46 _ThreadName=admin-thread-pool::admin-listener(1)] [timeMillis: 1540832141269] [levelValue: 1000] [[
  Import-Package: javax.servlet;javax.servlet.http
]]
[2018-10-29T16:55:41.270+0000] [Payara 5.184-SNAPSHOT] [SEVERE] [] [] [tid: _ThreadID=46 _ThreadName=admin-thread-pool::admin-listener(1)] [timeMillis: 1540832141270] [levelValue: 1000] [[
  DynamicImport-Package: *
]]
[2018-10-29T16:55:41.270+0000] [Payara 5.184-SNAPSHOT] [SEVERE] [] [] [tid: _ThreadID=46 _ThreadName=admin-thread-pool::admin-listener(1)] [timeMillis: 1540832141270] [levelValue: 1000] [[
  Created-By: 1.8.0_181-b02 (Azul Systems, Inc.)
]]
[2018-10-29T16:55:41.285+0000] [Payara 5.184-SNAPSHOT] [SEVERE] [NCLS-CORE-00032] [javax.enterprise.system.core] [tid: _ThreadID=46 _ThreadName=admin-thread-pool::admin-listener(1)] [timeMillis: 1540832141285] [levelValue: 1000] [[
  Error while expanding archive file
java.io.IOException: Stream closed
	at java.util.zip.ZipInputStream.ensureOpen(ZipInputStream.java:67)
	at java.util.zip.ZipInputStream.closeEntry(ZipInputStream.java:139)
	at org.glassfish.extras.osgicontainer.OSGiArchiveHandler.expandJar(OSGiArchiveHandler.java:241)
	at org.glassfish.extras.osgicontainer.OSGiArchiveHandler.expand(OSGiArchiveHandler.java:151)
	at com.sun.enterprise.v3.server.ApplicationLifecycle.getContext(ApplicationLifecycle.java:1937)
	at com.sun.enterprise.v3.server.ApplicationLifecycle.access$200(ApplicationLifecycle.java:118)
	at com.sun.enterprise.v3.server.ApplicationLifecycle$DeploymentContextBuidlerImpl.build(ApplicationLifecycle.java:1799)
	at org.glassfish.deployment.admin.DeployCommand.execute(DeployCommand.java:473)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl$2$1.run(CommandRunnerImpl.java:549)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl$2$1.run(CommandRunnerImpl.java:545)
	at java.security.AccessController.doPrivileged(Native Method)
	at javax.security.auth.Subject.doAs(Subject.java:360)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl$2.execute(CommandRunnerImpl.java:544)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl$3.run(CommandRunnerImpl.java:575)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl$3.run(CommandRunnerImpl.java:567)
	at java.security.AccessController.doPrivileged(Native Method)
	at javax.security.auth.Subject.doAs(Subject.java:360)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl.doCommand(CommandRunnerImpl.java:566)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl.doCommand(CommandRunnerImpl.java:1475)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl.access$1300(CommandRunnerImpl.java:111)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl$ExecutionContext.execute(CommandRunnerImpl.java:1857)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl$ExecutionContext.execute(CommandRunnerImpl.java:1733)
	at org.glassfish.admin.rest.resources.admin.CommandResource.executeCommand(CommandResource.java:409)
	at org.glassfish.admin.rest.resources.admin.CommandResource.execCommandSimpInMultOut(CommandResource.java:236)
	at sun.reflect.GeneratedMethodAccessor172.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.glassfish.jersey.server.model.internal.ResourceMethodInvocationHandlerFactory.lambda$static$0(ResourceMethodInvocationHandlerFactory.java:76)
	at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher$1.run(AbstractJavaResourceMethodDispatcher.java:148)
	at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.invoke(AbstractJavaResourceMethodDispatcher.java:191)
	at org.glassfish.jersey.server.model.internal.JavaResourceMethodDispatcherProvider$ResponseOutInvoker.doDispatch(JavaResourceMethodDispatcherProvider.java:200)
	at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.dispatch(AbstractJavaResourceMethodDispatcher.java:103)
	at org.glassfish.jersey.server.model.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:493)
	at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:415)
	at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:104)
	at org.glassfish.jersey.server.ServerRuntime$1.run(ServerRuntime.java:277)
	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:272)
	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:268)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:316)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:298)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:268)
	at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:289)
	at org.glassfish.jersey.server.ServerRuntime.process(ServerRuntime.java:256)
	at org.glassfish.jersey.server.ApplicationHandler.handle(ApplicationHandler.java:704)
	at org.glassfish.jersey.grizzly2.httpserver.GrizzlyHttpContainer.service(GrizzlyHttpContainer.java:377)
	at org.glassfish.admin.rest.adapter.JerseyContainerCommandService$3.service(JerseyContainerCommandService.java:179)
	at org.glassfish.admin.rest.adapter.RestAdapter.service(RestAdapter.java:182)
	at com.sun.enterprise.v3.services.impl.ContainerMapper$HttpHandlerCallable.call(ContainerMapper.java:516)
	at com.sun.enterprise.v3.services.impl.ContainerMapper.service(ContainerMapper.java:213)
	at org.glassfish.grizzly.http.server.HttpHandler.runService(HttpHandler.java:182)
	at org.glassfish.grizzly.http.server.HttpHandler.doHandle(HttpHandler.java:156)
	at org.glassfish.grizzly.http.server.HttpServerFilter.handleRead(HttpServerFilter.java:218)
	at org.glassfish.grizzly.filterchain.ExecutorResolver$9.execute(ExecutorResolver.java:95)
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeFilter(DefaultFilterChain.java:260)
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeChainPart(DefaultFilterChain.java:177)
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.execute(DefaultFilterChain.java:109)
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.process(DefaultFilterChain.java:88)
	at org.glassfish.grizzly.ProcessorExecutor.execute(ProcessorExecutor.java:53)
	at org.glassfish.grizzly.nio.transport.TCPNIOTransport.fireIOEvent(TCPNIOTransport.java:524)
	at org.glassfish.grizzly.strategies.AbstractIOStrategy.fireIOEvent(AbstractIOStrategy.java:89)
	at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy.run0(WorkerThreadIOStrategy.java:94)
	at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy.access$100(WorkerThreadIOStrategy.java:33)
	at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy$WorkerThreadRunnable.run(WorkerThreadIOStrategy.java:114)
	at org.glassfish.grizzly.threadpool.AbstractThreadPool$Worker.doWork(AbstractThreadPool.java:569)
	at org.glassfish.grizzly.threadpool.AbstractThreadPool$Worker.run(AbstractThreadPool.java:549)
	at java.lang.Thread.run(Thread.java:748)
]]
[2018-10-29T16:55:41.286+0000] [Payara 5.184-SNAPSHOT] [SEVERE] [] [javax.enterprise.system.core] [tid: _ThreadID=46 _ThreadName=admin-thread-pool::admin-listener(1)] [timeMillis: 1540832141286] [levelValue: 1000] [[
  Stream closed]]
[2018-10-29T16:55:42.289+0000] [Payara 5.184-SNAPSHOT] [SEVERE] [] [] [tid: _ThreadID=303 _ThreadName=Thread-67] [timeMillis: 1540832142289] [levelValue: 1000] [[
  Exception in thread "Thread-67" ]]
[2018-10-29T16:55:42.294+0000] [Payara 5.184-SNAPSHOT] [SEVERE] [] [] [tid: _ThreadID=303 _ThreadName=Thread-67] [timeMillis: 1540832142294] [levelValue: 1000] [[
  java.lang.RuntimeException: java.io.IOException: Pipe closed
	at org.glassfish.osgiweb.WebBundleURLStreamHandlerService$2.visit(WebBundleURLStreamHandlerService.java:226)
	at org.glassfish.osgijavaeebase.JarHelper.accept(JarHelper.java:74)
	at org.glassfish.osgiweb.WebBundleURLStreamHandlerService.writeWithoutSignedFiles(WebBundleURLStreamHandlerService.java:207)
	at org.glassfish.osgiweb.WebBundleURLStreamHandlerService.writeWithoutSignedFiles(WebBundleURLStreamHandlerService.java:196)
	at org.glassfish.osgiweb.WebBundleURLStreamHandlerService.access$100(WebBundleURLStreamHandlerService.java:90)
	at org.glassfish.osgiweb.WebBundleURLStreamHandlerService$1$1.run(WebBundleURLStreamHandlerService.java:128)
Caused by: java.io.IOException: Pipe closed
	at java.io.PipedInputStream.checkStateForReceive(PipedInputStream.java:260)
	at java.io.PipedInputStream.receive(PipedInputStream.java:226)
	at java.io.PipedOutputStream.write(PipedOutputStream.java:149)
	at java.util.zip.DeflaterOutputStream.deflate(DeflaterOutputStream.java:253)
	at java.util.zip.ZipOutputStream.closeEntry(ZipOutputStream.java:255)
	at org.glassfish.osgiweb.WebBundleURLStreamHandlerService$2.visit(WebBundleURLStreamHandlerService.java:224)
	... 5 more
]]
```
Seems to be an issue with opening a zip file o.0

**Comment by svendiedrichsen** [comment]
I think I have fixed it by reverting my changes to the FileUtils copy method.

**Comment by Cousjava** [comment]
Jenkins test please

**Comment by arjantijms** [comment]
@svendiedrichsen 
>I'd advise against forcing the merge. I'd rather analyse the problem first.

Yeah, I thought originally the tests passes and it was just the Jenkins job that disappeared before giving the okay signal.

**Comment by arjantijms** [comment]
Internally tracked by: PAYARA-3214

**Pull request merged**